MorseCode Decoder

inesmaio opened the pull request:
New feature.
Decoder from morse code into words and the other way around.
ESLint checked
Jest tests

inesmaio committed:
morseCode

First Commit
Validate if the input is morse code or words
Decode from morse code to words
Decode from words to morse code

inesmaio committed:
Lint

Lint enhancement

inesmaio committed:
EsLint config

Added Lint rules

inesmaio committed:
Merge branch 'master' into MorseCode

inesmaio committed:
Morse Code to Words

Morse Code to Words

inesmaio committed:
Lint enhancement

inesmaio committed:
Words to MorseCode - Tests - Lint

Added Words to MorseCode functions,
Added tests using jest
Run ESLint.

inesmaio committed:
README

Added README file.

bcotrim commented on .eslintrc.json:
```diff
@@ -17,5 +17,6 @@
     },
     "rules": {
         "no-plusplus": "off"
+        
```
Extra white line for no reason.
Seems like leftover form a change, I think you can delete this.

bcotrim commented on morseCode/morseCodeToWords.js:
```diff
@@ -0,0 +1,36 @@
+const morseCodeToWords = (arr) => {
+  let result = [];
+  const morseKeys = Object.keys(morseCode);
+  arr.map((elem) => {
```
I think `reduce` would be a better option for this.

bcotrim commented on morseCode/morseCodeToWords.test.js:
```diff
@@ -0,0 +1,9 @@
+const { mCTWordsResult } = require('./morseCodeToWords');
```
We are missing some test scenarios (null input for example)

Also it would be better to throw an error when getting an `undefined` translation instead of returning that result.

inesmaio replied:
The test case that returned undefined was removed, that it's not a real scenario, since the only way to get errors is if the input is incorrect, and I build a function to check the input before go through the decode execution.

bcotrim commented on morseCode/wordsToCodeMorse.js:
```diff
@@ -0,0 +1,35 @@
+// Decode input into morseCode
+const wordsToMorseCode = (arr) => {
+  let result = [];
+  arr.map((elem) => {
```
Same as before I think it would be better to use `reduce` for this case.

bcotrim commented on morseCode/validateInput.js:
```diff
@@ -0,0 +1,23 @@
+const { letters } = require('../utils/variables');
+
+
+const inputValidation = (userInput) => {
```
This is not only doing input validation.
It's also calling the appropriate function to solve the problem.

I think the function is good, just the name shouldn't be `inputValidation` in my opinion.

inesmaio replied:
You're right so, I changed the output of `inputValidation()`  to true or false.
This way I can only test the input.

inesmaio replied:
```suggestion
const inputValidation = (input) => {
  let validation;
  const REGEX = /[.-]/gm;
  if (input.search(REGEX) >= 0) {
    if (input.search(LETTERS) >= 0) {
      throw new InputError('Insert a valid Morse Code. No letter or numbers allowed.');
    } else {
      validation = true;
      return validation;
    }
  } else {
    validation = false
    return validation;
  }
};
```

inesmaio commented on morseCode/morseCodeToWords.js:
```diff
@@ -0,0 +1,36 @@
+const arrMorseCodeToWords = (input) => {
+  const SEPARATOR = '   ';
+  const inputArr = input.split(SEPARATOR);
+  const coderMorseArr = inputArr.map((element) => (element.split(' ')));
```
makes sence, thanks for the feedback

inesmaio committed:
Improvements

Changed morseToWord(), 
Changed wordToMorse(),
Changed inputValidation(),
Added and replace some tests.
